transcom/mymove
# INT-B-20001

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A949639&RoomContext=TeamRoom%3A848240&concept=TeamRoom)

## [Original Pull Request](https://github.com/transcom/mymove/pull/13228)

## Summary

PPTAS Reports requires Postman to test

`model_to_payload` requires the most reviewing and it's collapsed by default so don't forget to open it up!

### How to test

1. Import PPTAS api into Postman
2. Create a Navy move with an HHG and complete the entire move/payment process
3. http://primelocal:3000/pptas/v1/moves?since=2021-07-23T18:30:47.116Z
4. In DBeaver open the client_certs table
    look for the entry with `devlocal` in the subject column and update that row to `allow_pptas`
    the subject column will look something like this `/C=US/ST=DC/L=Washington/O=Truss/OU=AppClientTLS/CN=devlocal`
5. Check the report has the required details from the sheet linked below

> [!IMPORTANT]  
> Make sure you're using a valid TAC. You must also update the dates in the TAC and LOA tables to include this year.
> All dates should be within `lines_of_accounting.loa_bgn_dt` and `lines_of_accounting.loa_end_dt`
> All dates should be within `transportation_accounting_codes.trnsprtn_acnt_bgn_dt` and `transportation_accounting_codes.trnsprtn_acnt_end_dt`

You can use the queries below to update the dates for `E12A` tac line. Make sure you update the orders in the move to use the right tac

`UPDATE transportation_accounting_codes SET trnsprtn_acnt_end_dt = date('2026-01-01') WHERE tac = 'E12A'`
`update lines_of_accounting set loa_end_dt = date('2026-01-01') where id = '06254fc3-b763-484c-b555-42855d1ad5cd'`

[PPTAS working spreadsheet (27 June 2024) Jays Comments Dan Notes.xlsx](https://github.com/user-attachments/files/16183536/PPTAS.working.spreadsheet.27.June.2024.Jays.Comments.Dan.Notes.xlsx)
